### PR TITLE
Avoid word wrap crash when issue code long

### LINKED ISSUE #51: Summary fails for larger input
I am running the summary on multiple  sarif files and it seems that it does not correctly handle larger input

```
Run sarif summary ./*_scan.sarif -o ./hdf/issues.txt
Traceback (most recent call last):
  File "/home/runner/.local/bin/sarif", line 8, in <module>
    sys.exit(main())
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/cmdline/main.py", line 61, in main
    exitcode = args.func(args)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/cmdline/main.py", line 401, in _summary_command
    summary_op.generate_summary(input_files, output, multiple_file_output)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/operations/summary_op.py", line 39, in generate_summary
    summary_lines = _generate_summary(input_files)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/operations/summary_op.py", line 64, in _generate_summary
    issue_type_histogram = report.get_issue_type_histogram_for_severity(severity)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/issues_report.py", line 138, in get_issue_type_histogram_for_severity
    self._group_records_by_key()
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/issues_report.py", line 45, in _group_records_by_key
    key = combine_record_code_and_description(record)
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/sarif_file_utils.py", line 52, in combine_record_code_and_description
    return combine_code_and_description(record["Code"], record["Description"])
  File "/home/runner/.local/lib/python3.10/site-packages/sarif/sarif_file_utils.py", line 33, in combine_code_and_description
    shorter_description = textwrap.shorten(
  File "/usr/lib/python3.10/textwrap.py", line 414, in shorten
    return w.fill(' '.join(text.strip().split()))
  File "/usr/lib/python3.10/textwrap.py", line 371, in fill
    return "\n".join(self.wrap(text))
  File "/usr/lib/python3.10/textwrap.py", line 362, in wrap
    return self._wrap_chunks(chunks)
  File "/usr/lib/python3.10/textwrap.py", line 263, in _wrap_chunks
    raise ValueError("placeholder too large for max width")
ValueError: placeholder too large for max width
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by balgillo**
Add defensive code to avoid trying to run word wrap when there is not enough space on the line due to long issue code.  Also added unit test for this function `combine_code_and_description` that reproduced the issue.

Fixes #51

### CODE REVIEW

File: tests/test_sarif_file_utils.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    )
+    assert (
+        cd
+        == "ABC123 blah1 blah2 blah3 blah4 blah5 blah6 blah7 blah8 blah9 blah10 blah11 blah12 blah13 blah14 blah15 blah16 ..."
+    )
+    assert len(cd) <= 120
+    long_code = "".join(f"A{i}" for i in range(1, 36)) + "BC"
+    assert (
+        len(long_code) == 98
+    ), "98 is right length to hit'placeholder too large for max width' without defensive code"
```
Comment (debonte) [inline tests/test_sarif_file_utils.py:21]:
```suggestion
    ), "98 is right length to hit 'placeholder too large for max width' without defensive code"
```

File: tests/test_sarif_file_utils.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from sarif import sarif_file_utils
+
+
+def test_combine_code_and_description():
```
Comment (debonte) [inline tests/test_sarif_file_utils.py:4]: Personal preference perhaps, but I think this would be more readable if split into multiple test methods or if there was at least whitespace between the scenarios.

File: sarif/sarif_file_utils.py [hunk 3]
```diff
@@ -23,7 +23,9 @@ def combine_code_and_description(code: str, description: str) -> str:
         code = code.strip()
         length_budget -= len(code) + 1
     # Allow extra space when truncating for continuation characters
-    length_budget_pre_continuation = length_budget - 20
+    length_budget_pre_continuation = length_budget - 5
```
Comment (debonte) [inline sarif/sarif_file_utils.py:26]: What is the magic number 5? Maybe `" ..."` should be a constant and `5` should be calculated based on its length?

File: sarif/sarif_file_utils.py [hunk 4]
```diff
@@ -23,7 +23,9 @@ def combine_code_and_description(code: str, description: str) -> str:
         code = code.strip()
         length_budget -= len(code) + 1
```
Comment (debonte) [inline sarif/sarif_file_utils.py:24]: Why `+ 1`?

File: sarif/sarif_file_utils.py [hunk 5]
```diff
@@ -23,7 +23,9 @@ def combine_code_and_description(code: str, description: str) -> str:
...
     # Allow extra space when truncating for continuation characters
-    length_budget_pre_continuation = length_budget - 20
+    length_budget_pre_continuation = length_budget - 5
+    if length_budget_pre_continuation < 10:
```
Comment (debonte) [inline sarif/sarif_file_utils.py:27]: What is the magic number 10? Do we somehow know that it's impossible for `description` to be less than 10 characters, and therefore don't bother even looking at `description` in this case?

File: sarif/sarif_file_utils.py [hunk 6]
```diff
@@ -33,15 +35,15 @@ def combine_code_and_description(code: str, description: str) -> str:
             shorter_description = textwrap.shorten(
                 description, width=length_budget_pre_continuation, placeholder=" ..."
             )
-            if len(shorter_description) < 40:
+            if len(shorter_description) < length_budget_pre_continuation - 40:
```
Comment (debonte) [inline sarif/sarif_file_utils.py:38]: What is the magic number `40` here? Should it be `4` (i.e. the length of `" ..."`)?

File: sarif/sarif_file_utils.py (hunk 5 above)
Comment (balgillo) [reply sarif/sarif_file_utils.py:27]: It's just an approximation of the minimum length of a description to be useful, in the strange circumstances of a very long issue code.  I've added a comment to clarify.

File: tests/test_sarif_file_utils.py (hunk 2 above)
Comment (balgillo) [reply tests/test_sarif_file_utils.py]: Good point, I've split this into three tests

**Comment by balgillo** [comment]
Many thanks @debonte for reviewing.  I've implemented your suggestions and there are less magic numbers in this cosmetic formatting code now.  Please approve or let me know any other comments you have

**Review by debonte — APPROVED** [review]
